### PR TITLE
add steel bar door and trapdoor

### PR DESCRIPTION
**Pull request by TumeniNodes**
Just another change I made under my own preference, and figured I would put it up, if it might be desired / fit into MTG.
 These work great with dungeons, but look good in any situation.
It also expands door options a bit, which end users will probably enjoy.

screenshot_20190813_004449

screenshot_20190813_012142

I used xpanes due to the fact that using default:steel_ingots will not work, because of how the recipes are to make the steel bars.
Plus, making the doors from xpanes:bars makes sense.

### CODE REVIEW

File: mods/doors/init.lua [hunk 1]
```diff
@@ -505,6 +505,21 @@ doors.register("door_obsidian_glass", {
 		},
 })
 
+doors.register("door_bar", {
+		tiles = {{ name = "doors_door_bar.png", backface_culling = true }},
```
Comment (DS-Minetest) [inline mods/doors/init.lua]:
```suggestion
	tiles = {{ name = "doors_door_bar.png", backface_culling = true }},
```
\* 12

File: mods/doors/init.lua [hunk 2]
```diff
@@ -700,6 +715,19 @@ doors.register_trapdoor("doors:trapdoor_steel", {
 	groups = {cracky = 1, level = 2, door = 1},
 })
 
+doors.register_trapdoor("doors:trapdoor_bar", {
+	description = "Steel Bar Trapdoor",
+	inventory_image = "doors_trapdoor_bar.png",
+	wield_image = "doors_trapdoor_bar.png",
+	tile_front = "doors_trapdoor_bar.png",
+	tile_side = "doors_trapdoor_bar_side.png",
+	protected = true,
```
Comment (DS-Minetest) [inline mods/doors/init.lua]: - Why is this protected while the door is not?
- I wonder if protection is good for these doors and trapdoors. One the one hand you can grab through the bars, on the other hand who cares about realism, these doors might loog very good for jails and there might be a lock making grabbing through useless.

**Comment by DS-Minetest** [comment]
Doors (and maybe fencegate-like doors) that are like the xpanes bars in the middle of the node to align to them are needed.

**Comment by TumeniNodes** [comment]
> Doors (and maybe fencegate-like doors) that are like the xpanes bars in the middle of the node to align to them are needed.

Well, there is my centered doors mod. I will add these doors to that later as well.
In the bottom image I used the trap doors above the doors and then opened them, which does not look to bad.
If I were to add a fencegate type for this, it will require a diff model than the one existing. I don't mind doing it if people feel/agree it would be a nice addition and option added to gates.
But that does not fit with this PR

**Comment by TumeniNodes** [comment]
Dammit, hold on, I messed myself up when I added that commit to resolve... grrrrr. Curse you GIT!
I may need to redo this PR

**Comment by TumeniNodes** [comment]
I'm going to redo this PR in a couple hours, I have an appt I need to get to.

I want to know ahead of time, is it desired that I add 'protected' to the wooded trapdoor?

**Comment by TumeniNodes** [comment]
@DS-Minetest  I did not remove the tab in the tiles line at 509 as it does not maintain the style throughout the doors mod.
I did remove the spaces between the brackets though.
I am hoping the force push resolved the mess I made earlier, can you confirm if it did? It looks ok but the git is not strong with me

**Comment by paramat** [comment]
These look good.
I prefer that these are not crafted from xpanes items, to avoid a dependency on a minor specialist mod (which makes doors mod more difficult to use in other games), better to craft directly from steel ingots i think.

**Comment by paramat** [comment]
> Doors (and maybe fencegate-like doors) that are like the xpanes bars in the middle of the node to align to them are needed.

Well, ideally maybe, but not in MTG. We can consider xpane bars and these doors different and not for aligning.
> I want to know ahead of time, is it desired that I add 'protected' to the wooded trapdoor?

We can't if it has been non-protected until now. For lockable trapdoors we have steel ones.

**Comment by SmallJoker** [comment]
I like the door, and also the trapdoor is a nice addition. For me it would make most sense to add ownership/protection to doors, but not trapdoors. Former are usually used in prisons and can be locked, latter rather for street drains.

**Comment by paramat** [comment]
> For me it would make most sense to add ownership/protection to doors, but not trapdoors

I'm fine with that.

**Comment by TumeniNodes** [comment]
Ok Ill take care of these when I get home.
My only concern w using ingots was having doors cost less than a pane, but Ill take a closer look when I get back

I do agree adding a second hard dep is not good so I will fix that

**Comment by DS-Minetest** [comment]
Hm, the indentation in the doors mod is weird.
I'm not able to find any more code style issues.

Maybe it would be good adding these doors and trapdoors in the xpanes mod. No bar doors without bar xpanes.
Imo the recipe using xpanes is most intuitive and makes sense. If there's a dependency problem, it's still possible to use optional dependencies.

**Comment by paramat** [comment]
Yeah, maybe we can change the xpane bars recipe too if that makes sense.
> Maybe it would be good adding these doors and trapdoors in the xpanes mod.

Yes they could be registered in xpanes, but i would prefer them to be in doors mod, as i see xpanes as a minor specialist mod (which i would prefer not be in MTG, vessels mod too) and i see bar doors/trapdoors as being more fundamental, belonging with the other doors.

File: mods/doors/init.lua [hunk 3]
```diff
@@ -505,6 +505,21 @@ doors.register("door_obsidian_glass", {
...
+		inventory_image = "doors_item_bar.png",
+		groups = {cracky = 1, level = 2},
+		sounds = default.node_sound_metal_defaults(),
+		sound_open = "doors_steel_door_open",
+		sound_close = "doors_steel_door_close",
+		recipe = {
+			{"xpanes:bar_flat", "xpanes:bar_flat"},
+			{"xpanes:bar_flat", "xpanes:bar_flat"},
+			{"xpanes:bar_flat", "xpanes:bar_flat"},
+		},
```
Comment (paramat) [inline mods/doors/init.lua]: 12 lines above: only 1 indent needed.

**Comment by TumeniNodes** [comment]
>Yes they could be registered in xpanes, but i would prefer them to be in doors mod, as i see xpanes as a minor specialist mod (which i would prefer not be in MTG, vessels mod too) and i see bar doors/trapdoors as being more fundamental, belonging with the other doors.

ok, this paragraph just confused the hell out of me.
Please clarify.
Am I keeping these in doors and using steel ingots or
keeping in doors and using xpanes or
registering them _in_ xpanes (which I am not thrilled about tbh)

Here is my thinking:
To make the bars in xpanes should only take _maybe_ two steel ingots
To make the trapdoor = 4 steel ingots
To make the door = 6 steel ingots...
And keep these doors in the doors mod (where they belong)

Most importantly, will the changes to making xpanes bar_flat cause any issues in existing worlds/servers and if not, where/how do we get the new info to end users, etc.?
Obviously some documentation will need to be changed/updated

**Comment by TumeniNodes** [comment]
ok, a change of plan.
steel doors take 6 steel ingots, so rightfully steel bar doors should take less.

steel door takes 6 steel ingots
steel bar door should require 4 steel ingots
steel bar trapdoor can cost 2 steel ingots
and finally the xpanes:bar_flat changed to cost 1 steel ingot?

Would this layout be ok?

**Comment by TumeniNodes** [comment]
Redoing

File: mods/doors/init.lua (hunk 3 above)
Comment (DS-Minetest) [reply mods/doors/init.lua]: I already complaint about this. But elsewhere in this file, the indentation is the same.

**Pull request closed without merging**